models/gpg_key.go

```diff
@@ -759,17 +760,41 @@ func verifyWithGPGSettings(gpgSettings *git.GPGSettings, sig *packet.Signature,
 }
 
 // ParseCommitsWithSignature checks if signaute of commits are corresponding to users gpg keys.
-func ParseCommitsWithSignature(oldCommits *list.List) *list.List {
+func ParseCommitsWithSignature(oldCommits *list.List, repository *Repository) *list.List {
 	var (
 		newCommits = list.New()
 		e          = oldCommits.Front()
 	)
+	memberMap := map[int64]bool{}
+
 	for e != nil {
 		c := e.Value.(UserCommit)
-		newCommits.PushBack(SignCommit{
+		signCommit := SignCommit{
 			UserCommit:   &c,
 			Verification: ParseCommitWithSignature(c.Commit),
-		})
+		}
+
+		if signCommit.Verification.Verified {
+			signCommit.Verification.TrustStatus = "trusted"
+			if signCommit.Verification.SigningUser.ID != 0 {
+				isMember, has := memberMap[signCommit.Verification.SigningUser.ID]
+				if !has {
+					// We can ignore the error here as isMember would return false and so the user would be listed as untrusted
+					isMember, _ = repository.IsOwnerMemberCollaborator(signCommit.Verification.SigningUser.ID)
```

Comment:
This is certainly an improvemente. We should however consider that user membership is dynamic. One day a trusted user's commit might become untrusted because the user exits the project. Signatures become invalid because of expiration dates. In the future we should store the results of this check, IMHO.

Reply:
I disagree. If we store the results of the check then you have to somehow invalidate it.

If the user exits the project their commits may need to be untrusted and should probably be re-signed. There's actually no way of determining when a commit actually was - all of the data in it can be spoofed. The only trust you have is shared history or a signature. 

We should probably have a repository list of pubkeys that are trusted - perhaps stored in the repo itself although I'm not sure how to do the validation dance safely - this would be better than trusting our users randomly updated keys. 

In any case these levels are at present not tied into signed commits logic and only act as a visible warning to the user merging that there may be something amiss.

Reply:
Yeah, I guess there's no good way out, here. It will mainly depend on what the viewer is expecting from the information that is presented to them.